Automattic/wp-calypso
# Gutenboarding: Use domain categories from API.

#### Changes proposed in this Pull Request

Implements addition of domain suggestion categories.

See  D42138-code and D42551-code.

- [x] Add `category_slug` from D42551-code to the suggestions query.

Currently results look like

```jsonc
[
	{ title: 'Art & Design', slug: 'arts', tier: null },
	{ title: 'Business', slug: 'business', tier: null },
	{ title: 'Community', slug: 'community', tier: null },
	{ title: 'Education', slug: 'education', tier: null },
	{ title: 'Entertainment', slug: 'entertainment', tier: null },
	{ title: 'Fashion & Beauty', slug: 'fashion_beauty', tier: null },
	{ title: 'Financial', slug: 'financial', tier: null },
	{ title: 'Food & Drink', slug: 'food_drink', tier: null },
	{ title: 'Generic/Personal', slug: 'generic_personal', tier: null },
	{ title: 'Health & Wellness', slug: 'health_wellness', tier: null },
	{ title: 'International', slug: 'international', tier: 2 },
	{ title: 'Marketing & Media', slug: 'marketing_media', tier: null },
	{ title: 'Popular', slug: 'popular', tier: 1 },
	{ title: 'Professional', slug: 'professional', tier: null },
	{ title: 'Real Estate & Home', slug: 'real_estate_home', tier: null },
	{ title: 'Shopping & Commerce', slug: 'shopping_commerce', tier: null },
	{ title: 'Sports & Recreation', slug: 'sports_rec', tier: null },
	{ title: 'Technology', slug: 'technology', tier: null },
	{ title: 'Travel & Transit', slug: 'travel_transit', tier: null },
]
```

cc: @daledupreez 

#### Testing instructions

1. On `/new`, run this in `wp.data.select('automattic/domains/suggestions').getCategories()` in console TWICE.
    - You should see an empty array the first time.
    - Inspect network panel for the request being made.
    - You should see an array of 19 domain categories on the second time.

2. Open up domain picker modal (Domain Picker Button > More Options) and switch between different domain category on the sidebar.
    - It should load up new domain suggestions.
    - Inspect the network panel to see if `category_slug` param is being used for the domain suggestions API request.


#### Screenshots

![image](https://user-images.githubusercontent.com/1287077/82288292-767ee200-99a2-11ea-96fa-6453056fbbc2.png)
![image](https://user-images.githubusercontent.com/1287077/82288296-78e13c00-99a2-11ea-8196-c2856aed7375.png)
![image](https://user-images.githubusercontent.com/1287077/82288301-7b439600-99a2-11ea-9386-b23873bac4e4.png)
